src/rng.c

```diff
@@ -56,13 +56,13 @@ void rng_gen(unsigned char *buf, size_t buflen)
 			rc = fread(buf, buflen, 1, rng_fh);
 			fclose(rng_fh);
 			if (rc == 1)
-				return;
+				return 0;
 		}
 	}
 
-	syslog(LOG_ERR, "Libica internal RNG error..");
-	fprintf(stderr, "Libica internal RNG error.");
-	exit(1);
+	/* unable to get any random data. cleanup and return with error */
+	memset(buf, 0, buflen);
```

Comment:
I found a missing #include <string.h>, so the memset() usage causes a warning. I'll fix that issue.